data-transforms/flatten/transform.go

```diff
@@ -104,18 +103,45 @@ func descend(w io.Writer, kv jstream.KV, depth int, key string, delim string) {
 
 		for index, kv := range kvs {
 			new_key := key + delim + kv.Key
-			var isLastSubElementLocal = len(kvs) - 1 == index
 			descend(w, kv, depth+1, new_key, delim)
-			if ! isLastSubElementLocal {
-				fmt.Fprint(w, ",\n")
-			}
+			handleLastElement(w, len(kvs), index, ",\n")
 		}
-		// fallthrough
 	default:
 		if kv.Value != nil {
 			fmt.Fprintf(w, "  \"%s\": %v", key, kv.Value)
 		} else {
 			fmt.Fprintf(w, "  \"%s\": null", key)
 		}
 	}
-}
\ No newline at end of file
+}
+
+
+func handleLastElement(w io.Writer, listSize int, index int, endingString string){
+	var isLastSubElementLocal = listSize - 1 == index
+	if ! isLastSubElementLocal {
+		fmt.Fprintf(w, "%s", endingString)
+	}
+}
+
+
+func flattenListOfJsonObjects(w io.Writer, kv jstream.KV, depth int, key string, delim string) bool {
+	isNodeFlattened := false
+	for parent_index, v := range kv.Value.([]interface{}) {
+		switch v.(type) {
+		case jstream.KVS:
+			isNodeFlattened = true
+			kvs := v.(jstream.KVS)
+
+			for index, kv := range kvs {
+				new_key := key + delim + fmt.Sprint(parent_index) + delim + kv.Key
+				descend(w, kv, depth+1, new_key, delim)
+				handleLastElement(w, len(kvs), index, ",\n")
+			}
+			handleLastElement(w, len(kv.Value.([]interface{})), parent_index, ",\n")
+
+		default:
+			return isNodeFlattened
```

Comment:
Why are we returning early here? Does this end up skipping over elements if there are heterogeneous lists? I.E.

```json
[
  "foo",
  {"a":4},
  5,
]
```

I **think** the above case will drop the 5 value.

Reply:
The above doesn't drop values, but it doesn't flatten the `{"a": 4}` Object, either. If the first element (`"foo"`) is removed, it will drop the `5`.

I'm conflicted on what the correct approach should be here. My original intent was to provide a close to drop in replacement for the Kafka Connect Flatten SMT. I believe the Flatten SMT supports heterogeneous Array flattening, but requires a schema. Maybe that's the correct approach here?

Reply:
Sorry, I didn't have much time to put in this PR lately. It has been a moment since I opened this PR, I don't remember why, to be honest. I will look it up as well when I will have a moment.